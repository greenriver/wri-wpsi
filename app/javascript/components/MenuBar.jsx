import React from 'react'
import { NavLink } from 'react-router-dom'
import withPages from 'components/withPages'
import styleVariables from './styles/variables'

const MenuBar = ({ pages }) => {
  const containerStyle = {
    fontSize: 21,
    width: '100%',
  }

  const inactiveLinkStyle = {
    color: 'white',
    textDecoration: 'none',
    padding: 25,
    display: 'inline-block',
    fontWeight: 'normal',
    ':hover': {
      color: styleVariables().colors.primary,
    },
  }

  const activeLinkStyle = {
    color: styleVariables().colors.primary,
    fontWeight: '600',
  }

  const staticContentPageLinks = pages.map(
    page => (
      <NavLink
        to={`/info/${page.slug}`}
        key={page.slug}
        style={inactiveLinkStyle}
        activeStyle={activeLinkStyle}
      >
        {page.name}
      </NavLink>
    )
  )

  return <div style={containerStyle}>
    <NavLink
      to='/map'
      exact
      style={inactiveLinkStyle}
      activeStyle={activeLinkStyle}
    >
      Map
    </NavLink>

    {staticContentPageLinks}
  </div>
}

import PropTypes from 'prop-types'
MenuBar.propTypes = {
  pages: PropTypes.array.isRequired,
}

export default withPages(MenuBar)
