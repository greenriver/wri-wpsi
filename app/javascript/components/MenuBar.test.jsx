import React from 'react'
import { BrowserRouter, NavLink } from 'react-router-dom'
import { shallow } from 'enzyme';
import MenuBar from 'components/MenuBar'
import renderer from 'react-test-renderer'

describe('<MenuBar />', () => {
  it('renders correctly when inside a router', () => {
    const component = renderer.create(
      <BrowserRouter>
        <MenuBar></MenuBar>
      </BrowserRouter>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders three <NavLink /> components', () => {
    const component = shallow(<MenuBar />)
    expect(component.find(NavLink)).toHaveLength(3)
  })
})
