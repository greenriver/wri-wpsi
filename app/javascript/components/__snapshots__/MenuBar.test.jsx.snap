// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar /> renders correctly when inside a router 1`] = `
<div
  style={
    Object {
      "fontSize": 21,
      "margin": 9,
      "width": "100%",
    }
  }
>
  <a
    aria-current={null}
    href="/map"
    onClick={[Function]}
    style={
      Object {
        "color": "#999",
        "display": "inline-block",
        "fontWeight": "normal",
        "margin": 25,
        "textDecoration": "none",
      }
    }
  >
    Map
  </a>
  <a
    aria-current={null}
    href="/methodology"
    onClick={[Function]}
    style={
      Object {
        "color": "#999",
        "display": "inline-block",
        "fontWeight": "normal",
        "margin": 25,
        "textDecoration": "none",
      }
    }
  >
    Methodology
  </a>
  <a
    aria-current={null}
    href="/about"
    onClick={[Function]}
    style={
      Object {
        "color": "#999",
        "display": "inline-block",
        "fontWeight": "normal",
        "margin": 25,
        "textDecoration": "none",
      }
    }
  >
    About Us
  </a>
  <a
    href="/admin"
    style={
      Object {
        "color": "#999",
        "display": "inline-block",
        "float": "right",
        "fontWeight": "normal",
        "margin": 25,
        "textDecoration": "none",
      }
    }
  >
    Admin Portal
  </a>
</div>
`;
